test/e2e/leaderelection.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
```

Comment:
The test here seems to be more suitable as an in package test. The only dependency that I see used from `e2e framework` is the `config`, which really doesn't warrant usage of `e2e framework`.

Reply:
This test absolutely belongs in the e2e package.  It is dependent on the managed fixture (test-managed federation composed from kubeapiserver and etcd binaries.

Reply:
Yeah, understood the need for API server now.